### DISCUSSION TITLE
Converting over to native M1 Apple Silicon from /usr/local to /opt/homebrew - Best way?

### DISCUSSION THREAD

**Discussion by wakatara** [Homebrew/discussions#780] [Getting started]
Hey homebrewers,

I have an M! Mac but have been running homebrew in rosetta2 mode since I got the machine in late Nov.

Now M1 is officially supported I wanted to shift things over to native (I'll compile from source if necessary.).

What is the advised way to do this? Should I just uninstall the old `/usr/local` homebrew via the uninstall script (after grabbing a `brew list > ~/Desktop/brews.txt`) and then reinstall selecting the M1 native version and brew install the brew list? 

_(also, I just want to say thanks to all the maintainers providing such a ridiculously useful tool for everyone. Even when using verious linuxen with its slew of package managers, I'm often amazed at how fluid homebrew seems in comparison. Merci!)_

**Comment by carlocab** [comment]
You can use `brew bundle`. For example, `/usr/local/bin/brew bundle dump` will generate a `Brewfile` in the current directory containing all the information needed to reproduce your existing install. `/opt/homebrew/bin/brew bundle install` should install everything in your `Brewfile` in your current directory.

See also #707, #417, #437, #545, #357.

**Reply by wakatara** [reply]
Thanks heaps for this. I uninstalled completely and then manually reinstalled after the `brew list` piped to text I did. I did have to manually nuke `/usr/local` subs, but other than the stupidity of me deleting my fish shell and locking myself out of terminal for a few seconds, went very smooth and (while I may be imagining it) the M1 versions of everything seem peppier.

Oh, and needed of course to reinstall python libs and the like via pip3 but other than that, pretty smooth so far.

thanks for your help!!
